onnxruntime/core/framework/debug_node_inputs_outputs_utils.cc

```diff
@@ -231,19 +240,30 @@ const NodeDumpOptions& NodeDumpOptionsFromEnvironmentVariables() {
   return node_dump_options;
 }
 
+static bool AnyOptionIsSet(const NodeDumpOptions& dump_options) {
+  return ((dump_options.dump_flags & NodeDumpOptions::DumpFlags::ShapeOnly) != 0) ||
+         ((dump_options.dump_flags & NodeDumpOptions::DumpFlags::InputData) != 0) ||
+         ((dump_options.dump_flags & NodeDumpOptions::DumpFlags::OutputData) != 0);
+}
+
+#define COUT_IF(boolean_expression) \
```

Comment:
would prefer to avoid macros. this could be:
```
void PrintIf(bool condition, const std::string& message);
PrintIf(condition, MakeString(...));
```